test-data/unit/check-typeddict.test

```diff
@@ -813,3 +813,29 @@ p = TaggedPoint(type='2d', x=42, y=1337)
 p.get('x', 1 + 'y')     # E: Unsupported operand types for + ("int" and "str")
 [builtins fixtures/dict.pyi]
 [typing fixtures/typing-full.pyi]
+
+
+-- Special cases
```

Comment:
Just have few comments:
* Will this PR also fix #3316 and #2489?
* There is a similar situation with ``NamedTuple`` (it just uses ``fallback = self.named_type('__builtins__.tuple', [AnyType()])`` now).
* Maybe it makes sense to add class syntax to the tests just in case?